### PR TITLE
feat: create convo draft store

### PR DESCRIPTION
**Pull request by BlankParticle**
## What does this PR do?

Save drafts in localstore and load them in each convo and new convo form

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Required

- [ ] Read [Contributing Guide](https://github.com/un/inbox/blob/main/CONTRIBUTING.md)
- [ ] Self-reviewed my own code
- [ ] Tested my code in a local environment
- [ ] Commented on my code in hard-to-understand areas
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the UnInbox Docs if changes were necessary

### CODE REVIEW

**Comment by BlankParticle** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/un/inbox/636?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#636** <a href="https://app.graphite.dev/github/pr/un/inbox/636?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#635** <a href="https://app.graphite.dev/github/pr/un/inbox/635?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @BlankParticle and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by McPizza0** [comment]
i see changes still being pushed
let me know when its done so i can review

**Comment by BlankParticle** [comment]
> i see changes still being pushed let me know when its done so i can review

It was ready 30 min ago, I just changed pushed stuff which I forgot to push

**Review by McPizza0 — CHANGES_REQUESTED** [review]
we also need to save the org (and soon "space) that the draft is in
this is also just draft for new convos right?
and it only saves latest draft?

might need a way to clear the draft content from the email if I no longer want the draft

**Comment by BlankParticle** [comment]
Lets put this one in hold for a day

**Comment by BlankParticle** [comment]
Updated the Draft store to account for orgShortcode and spacecode when we add it.

I also had to remove atoms as keeping them in sync with draft was hard as orgShortCode is not known when the atom is initialized.

File: apps/web/src/app/[orgShortcode]/convo/_components/create-convo-form.tsx [hunk 1]
```diff
@@ -306,9 +344,29 @@ export default function CreateConvoForm({
     newEmailParticipants
   ]);
 
-  const [editorText, setEditorText] = useState<JSONContent>(
-    emptyTiptapEditorContent
-  );
+  const [editorText, setEditorText] = useState(draft.content);
+
+  // Autosave draft
+  const debouncedEditorText = useDebounce(editorText, 500);
```
Comment (McPizza0) [inline apps/web/src/app/[orgShortcode]/convo/_components/create-convo-form.tsx:350]: this debounce can be much higher
5 seconds even

Comment (BlankParticle) [reply apps/web/src/app/[orgShortcode]/convo/_components/create-convo-form.tsx:350]: that would have the issue of not saving your editor text of last 5 second if you navigate early

**Review by ashgansh — APPROVED** [review]

**Review by McPizza0 — APPROVED** [review]

**Comment by BlankParticle** [comment]
### Merge activity

* **Aug 7, 6:56 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/un/inbox/636) rebased this pull request as part of a merge.

**Pull request merged**